controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -83,6 +123,46 @@ bool wfa_ca::reply(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx, const std::st
     return message_com::send_cmdu(sd, cmdu_tx);
 }
 
+bool wfa_ca::parse_params(std::unordered_map<std::string, std::string> &params,
+                          std::string &err_string)
+{
+    // create copy of all mandatory param names
+    std::list<std::string> mandatory_params;
+    for (const auto &param_name : params) {
+        mandatory_params.push_back(param_name.first);
+    }
+
+    size_t mandatory_params_cnt = 0;
+    while (!finished_tokens()) {
+        std::string param_name = get_next_token();
+        std::transform(param_name.begin(), param_name.end(), param_name.begin(), ::tolower);
+        if (params.find(param_name) != params.end()) {
```

Comment:
I guess here you meant mandatory_params instead of params.

I don't think that will work well. I think it's better to pass mandatory_params explicitly as an input argument. Then you can make a copy of it (implicit because you don't pass it by reference) and just delete the ones that you find. That also solves the issue that counting is incorrect if the same parameter occurs twice.

Reply:
It works well, I tested it.
Your way will work as well. The only difference is how many arguments you have to the function. Mine has one less.
In my implementation, I can also delete mandatory params from the list and at the end of the function, check if the list is empty, but the counter is good too.
I don't see a reason to change it.

Reply:
True, you could delete it from the list, I didn't think of that.

Anyway, I should have put (nitpick) here.